File: madameautomata-bot-python-autoreg/bot/commands/bratjail.py

import discord
from discord import app_commands, Interaction
from ..theme import gothic_embed
from ..util.perm import is_domme, is_staff, is_sub
from ..db import Session, User, Collaring, Cage, JailSession, Star, Task, Worship, Config, CollarType, StarStatus
from ..util.time import parse_duration, random_case
from ..util.ids import cuid
from datetime import datetime, timedelta
from sqlalchemy import select, func, update, and_

def setup(bot):
    @bot.tree.command(name="bratjail", description="Harsher jail; mistakes add to total and case rerandomizes.")
    @app_commands.describe(sub="Submissive", sentence="The sentence they must type", times="How many times")
    async def bratjail(i: Interaction, sub: discord.User, sentence: str, times: int):
        member = i.guild.get_member(i.user.id)
        if not await is_domme(member):
            return await i.response.send_message("Only Dommes can use this.", ephemeral=True)
        seed = random.randint(1, 1_000_000_000)
        async with Session() as s:
            s.add(JailSession(id=cuid(), sub_id=sub.id, domme_id=i.user.id, brat=True, sentence_text=sentence, total_needed=times, case_seed=seed))
            await s.commit()
        await i.response.send_message(embed=gothic_embed("Brat Jail", f"**<@{sub.id}>** is in brat jail. Type this exactly **{times}** times (mistakes add 1 to total):\n\n`{random_case(sentence, seed)}`"))

File: madameautomata-js-bot/src/commands/bratjail.js

import { SlashCommandBuilder, PermissionFlagsBits, ChannelType, ActionRowBuilder, ButtonBuilder, ButtonStyle, StringSelectMenuBuilder, EmbedBuilder } from "discord.js";
import { prisma } from "../lib/prisma.js";
import { Theme, gothicEmbed } from "../theme/theme.js";
import { parseDuration, randomCase } from "../util/duration.js";
import { cuid } from "../util/ids.js";
import { isDomme, isSub, isStaff, getConfig } from "../util/perm.js";

const cmd = 
{
  data: new SlashCommandBuilder()
    .setName("bratjail")
    .setDescription("Harsher jail; mistakes add to total and case re-randomizes.")
    .addUserOption(o=>o.setName("sub").setDescription("Submissive").setRequired(true))
    .addStringOption(o=>o.setName("sentence").setDescription("Sentence to type").setRequired(true))
    .addIntegerOption(o=>o.setName("times").setDescription("How many times").setRequired(true)),
  async run(i) {
    if (!await isDomme(i.member)) return i.reply({ content: "Only Dommes can use this.", ephemeral: true });
    const sub = i.options.getUser("sub", true);
    const sentence = i.options.getString("sentence", true);
    const times = i.options.getInteger("times", true);
    const seed = Math.floor(Math.random()*1e9);
    await prisma.jailSession.create({ data: { id: cuid(), subId: sub.id, dommeId: i.user.id, brat: true, sentenceText: sentence, totalNeeded: times, caseSeed: seed, active: true } });
    return i.reply({ embeds: [gothicEmbed("Brat Jail", `**<@${sub.id}>** must type this exactly **${times}** times. Mistakes add +1 and re-randomize.`)] });
  }
}


export default cmd;
